x-pack/legacy/plugins/maps/public/elasticsearch_geo_utils.js

```diff
@@ -203,118 +235,134 @@ export function geoShapeToGeometry(value, accumulator) {
   accumulator.push(geoJson);
 }
 
-const POLYGON_COORDINATES_EXTERIOR_INDEX = 0;
-const TOP_LEFT_INDEX = 0;
-const BOTTOM_RIGHT_INDEX = 2;
+function createGeoBoundBoxFilter(geometry, geoFieldName, filterProps = {}) {
+  ensureGeometryType(geometry.type, [GEO_JSON_TYPE.POLYGON]);
+
+  const TOP_LEFT_INDEX = 0;
+  const BOTTOM_RIGHT_INDEX = 2;
+  const verticies = geometry.coordinates[POLYGON_COORDINATES_EXTERIOR_INDEX];
+  return {
+    geo_bounding_box: {
+      [geoFieldName]: {
+        top_left: verticies[TOP_LEFT_INDEX],
+        bottom_right: verticies[BOTTOM_RIGHT_INDEX]
+      }
+    },
+    ...filterProps
+  };
+}
 
 export function createExtentFilter(mapExtent, geoFieldName, geoFieldType) {
+  ensureGeoField(geoFieldType);
+
   const safePolygon = convertMapExtentToPolygon(mapExtent);
+
   if (geoFieldType === ES_GEO_FIELD_TYPE.GEO_POINT) {
-    const verticies = safePolygon.coordinates[POLYGON_COORDINATES_EXTERIOR_INDEX];
-    return {
-      geo_bounding_box: {
-        [geoFieldName]: {
-          top_left: verticies[TOP_LEFT_INDEX],
-          bottom_right: verticies[BOTTOM_RIGHT_INDEX]
-        }
+    return createGeoBoundBoxFilter(safePolygon, geoFieldName);
+  }
+
+  return {
+    geo_shape: {
+      [geoFieldName]: {
+        shape: safePolygon,
+        relation: ES_SPATIAL_RELATIONS.INTERSECTS
       }
-    };
-  } else if (geoFieldType === ES_GEO_FIELD_TYPE.GEO_SHAPE) {
+    }
+  };
+}
+
+export function createGeometryFilterWithMeta({
+  geometry,
+  geometryLabel,
+  indexPatternId,
+  geoFieldName,
+  geoFieldType,
+  relation = ES_SPATIAL_RELATIONS.INTERSECTS,
+  isBoundingBox = false,
```

Comment:
remove boolean flag as parameter, and instead create second  public function `createExtentFilterWithMeta`. Have external clients call this.

Internally, it can then delegate as appropriate.

Reply:
Why? That seems like a lot of extra wrapping and extra calls. You will have two functions with very similar signatures and end up just calling the same function. Then the caller has to have extra logic to call the right function.

The caller just wants to create a spatial filter. Whether that filter is a bounding box or not is just a parameter to creating a spatial filter and is an implementation detail that the caller should not need to know how it effects the filter. Having a separate function makes the caller need to have logic that should be internal to creating the filter.

Reply:
Discussed offline and will split interface into `createSpatialFilterWithGeometry` and `createSpatialFilterWithBoundingBox`